vdr/src/contracts/anoncreds/revocation_registry.rs

```diff
@@ -0,0 +1,845 @@
+pub async fn resolve_revocation_registry_definition(
+    client: &LedgerClient,
+    id: &RevocationRegistryDefinitionId,
+) -> VdrResult<RevocationRegistryDefinition> {
+    let parsed_id = ParsedRevocationRegistryDefinitionId::try_from(id)?;
+    match (parsed_id.network.as_ref(), client.network()) {
+        (Some(schema_network), Some(client_network)) if schema_network != client_network => {
+            return Err(VdrError::InvalidRevocationRegistryDefinition(format!("Network of request revocation registry definition id {} does not match to the client network {}", schema_network, client_network)));
+        }
+        _ => {}
+    };
+
+    let transaction = build_resolve_revocation_registry_definition_transaction(client, id).await?;
+    let response = client.submit_transaction(&transaction).await?;
+    if response.is_empty() {
+        return Err(VdrError::ClientInvalidResponse(format!(
+            "Revocation Registry Definition not found for id: {:?}",
+            id
+        )));
+    }
+    let rev_reg_def_record =
+        parse_resolve_revocation_registry_definition_result(client, &response)?;
+
+    let rev_reg_def_id = rev_reg_def_record.revocation_registry_definition.id();
+    if &rev_reg_def_id != id {
+        return Err(VdrError::InvalidRevocationRegistryDefinition(format!(
+            "Revocation Registry Definition ID {} does not match to requested {}",
+            rev_reg_def_id.to_string(),
+            id.to_string()
+        )));
+    }
+
+    Ok(rev_reg_def_record.revocation_registry_definition)
+}
+
+/// Single step function to resolve a Revocation Registry Status List for the given ID at a given
+/// timestamp
+///
+/// # Params
+/// - `client`: [LedgerClient] - client connected to the network where contract will be executed
+/// - `id`: [RevocationRegistryId] - id of Revocation Registry Definition to resolve
+/// - `to_timestamp`: [u64] - timestamp of the Revocation Registry Status List resolution
+///
+/// # Returns
+///   revocation_status_list: [RevocationStatusList] - Resolved Revocation Status List object
+pub async fn resolve_revocation_registry_status_list(
+    client: &LedgerClient,
+    id: &RevocationRegistryDefinitionId,
+    to_timestamp: u64,
+) -> VdrResult<RevocationStatusList> {
+    let rev_reg_def = resolve_revocation_registry_definition(&client, &id).await?;
+    let delta = fetch_revocation_delta(&client, &id, to_timestamp)
+        .await?
+        .ok_or(VdrError::InvalidRevocationRegistryStatusList(format!(
+            "No revocation status list found for Revocation Registry Definition ID {}",
+            id.as_ref()
+        )))?;
+
+    delta.validate(rev_reg_def.value.max_cred_num - 1)?;
+
+    let mut revocation_list: Vec<u32> = vec![0; rev_reg_def.value.max_cred_num.try_into().unwrap()];
+
+    // Set all `issuer` indexes to 0 (not revoked)
+    for issue in delta.issued {
+        revocation_list[issue as usize] = RevocationState::Active as u32
+    }
```

Comment:
I think this loop does nothing because `let mut revocation_list: Vec<u32> = vec![0; rev_reg_def.value.max_cred_num.try_into().unwrap()];` already create an array where all elements are `0`